core/src/test/kotlin/io/specmatic/core/ScenarioTest.kt

```diff
@@ -658,4 +662,254 @@ class ScenarioTest {
     }
+
+    @Nested
+    inner class CalculatePathTests {
+        @Test
+        fun `calculatePath should handle JSONObjectPattern body`() {
+            val scenario = Scenario(
+                name = "test",
+                httpRequestPattern = HttpRequestPattern(
+                    body = JSONObjectPattern(
+                        mapOf("data" to AnyPattern(listOf(StringPattern()))),
+                        typeAlias = "(TestRequest)"
+                    )
+                ),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = JSONObjectValue(mapOf("data" to StringValue("test")))
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactly("{TestRequest}.data{string}")
+        }
+
+        @Test
+        fun `calculatePath should handle AnyPattern body with scalar wrapping`() {
+            val scenario = Scenario(
+                name = "test",
+                httpRequestPattern = HttpRequestPattern(
+                    body = AnyPattern(listOf(StringPattern(), NumberPattern()))
+                ),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = StringValue("test")
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactly("{string}")
+        }
+
+        @Test
+        fun `calculatePath should handle AnyPattern body with non-scalar types`() {
+            val objectPattern = JSONObjectPattern(
+                mapOf("field" to StringPattern()),
+                typeAlias = "(CustomObject)"
+            )
+            val scenario = Scenario(
+                name = "test",
+                httpRequestPattern = HttpRequestPattern(
+                    body = AnyPattern(listOf(objectPattern))
+                ),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = JSONObjectValue(mapOf("field" to StringValue("test")))
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactly("CustomObject")
```

Comment:
This should be:

assertThat(paths).containsExactly("{CustomObject}")

CustomObject was a typeAlias, and hence should have appeared within braces.